test/unittest/tests/block/test_block_addr.cpp

```diff
@@ -0,0 +1,169 @@
+
+    unpack_addr_cookie_and_check(begin, block->allocsize, (wt_off_t)cookie_vals[0],
+      (uint32_t)cookie_vals[1], (uint32_t)cookie_vals[2]);
+}
+
+TEST_CASE("Block addr pack and unpack", "[block_addr]")
+{
+    WT_BLOCK b;
+    WT_BM bm, *bmp;
+
+    b.allocsize = 1;
+    bmp = &bm;
+    bmp->block = &b;
+
+    SECTION("Pack and unpack address cookie 1")
+    {
+        uint32_t pack_checksum, pack_size;
+        wt_off_t pack_offset;
+        pack_checksum = pack_size = 0;
+        pack_offset = 0;
+
+        // Test the block manager's pack function with an address cookie containing all zero fields.
+        test_pack_and_unpack_addr_cookie(bmp->block, pack_offset, pack_size, pack_checksum);
+    }
+
+    SECTION("Pack and unpack address cookie 2")
+    {
+        uint32_t pack_checksum, pack_size;
+        wt_off_t pack_offset;
+        pack_checksum = 1;
+        pack_size = 0;
+        pack_offset = 1;
+
+        /*
+         * Test that packing an address cookie of size 0 just packs 0 into all the fields. The pack
+         * values will differ from the expected values (which are all 0), and the function checks
+         * for this.
+         */
+        test_pack_and_unpack_addr_cookie(bmp->block, pack_offset, pack_size, pack_checksum);
+    }
+
+    SECTION("Pack and unpack address cookie 3")
+    {
+        uint32_t pack_checksum, pack_size;
+        wt_off_t pack_offset;
+        pack_checksum = 12345;
+        pack_offset = 10;
+        pack_size = 4;
+
+        // Test packing an address cookie with mostly non-zero fields.
+        test_pack_and_unpack_addr_cookie(bmp->block, pack_offset, pack_size, pack_checksum);
+    }
+
+    SECTION("Manually pack and unpack address cookie 4")
+    {
+        /*
+         * The block manager will modify these values due to its logic that accounts for large
+         * offsets. The address cookie values that will actually be packed with an allocsize of 1:
+         * {7, 7, 42}.
+         */
+        std::vector<int64_t> cookie_vals = {8, 7, 42};
+        std::vector<uint8_t> expected_packed_vals = {0x87, 0x87, 0xaa};
+        test_pack_and_unpack_addr_cookie_manual(bmp->block, cookie_vals, expected_packed_vals);
+    }
+
+    SECTION("Pack and unpack address cookie with negative values")
+    {
+        // Test that trying to pack an address cookie with negative values exhibits weird behavior.
+        std::vector<int64_t> cookie_vals = {-6, -42, -256};
+        std::vector<uint8_t> expected_packed_vals = {0x79, 0x56, 0x3f, 0x40};
+
+        std::vector<uint8_t> packed(24, 0);
+        uint8_t *p = packed.data();
+        REQUIRE(
+          __wt_block_addr_pack(bmp->block, &p, WT_TIERED_OBJECTID_NONE, (wt_off_t)cookie_vals[0],
+            (uint32_t)cookie_vals[1], (uint32_t)cookie_vals[2]) == 0);
+        CHECK(packed[0] != expected_packed_vals[0]);
+        CHECK(packed[1] != expected_packed_vals[1]);
+        CHECK(packed[2] != expected_packed_vals[2]);
+    }
```

Comment:
Could we check the case where objectid is non zero?

Reply:
I thought about checking this, but given that tiered storage is getting deprecated it feels less relevant. I can still add in a test case if it's desired though

Reply:
Good point, I am happy to not test it as well then.